gemrb/core/Map.cpp

```diff
@@ -2392,12 +2392,14 @@ void Map::PlayAreaSong(int SongType, bool restart, bool hard) const
 {
 	size_t pl = SongList[SongType];
 	const ieVariable* poi = &core->GetMusicPlaylist(pl);
-	// for subareas fall back to the main list
-	// needed eg. in bg1 ar2607 (intro candlekeep ambush south)
+
+	// Some subareas don't have their own songlist and GemRB would fail to play
+	// the expected music then. Currently this is only observed for the battle
+	// music in BG1, e.g. AR2607 (intro candlekeep ambush south).
+	// In this case GemRB falls back to the music referenced directly by SongType.
 	// it's not the correct music, perhaps it needs the one from the master area
 	// it would match for ar2607 and ar2600, but very annoying (see GetMasterArea)
-	// ... but this is also definitely wrong for iwd
-	if (IsStar(*poi) && !MasterArea && SongType == SONG_BATTLE && core->HasFeature(GFFlags::BREAKABLE_WEAPONS)) {
+	if (IsStar(*poi) && !MasterArea && SongType == SONG_BATTLE && core->config.GameType == "bg1") {
```

Comment:
1. This is not better, we never check by game type. The proper fix is to add another game flag, if one is needed, which is still unclear.
2. The iwd comment is valuable testing info, why did you remove it?

Reply:
Both of your points result from me reading to much into the existing comment and test for the `BREAKABLE_WEAPON' flag. I thought that it is indeed kind of established that this is only relevant for BG1 and I could condense the comment away and use a more direct gamecheck.

Now I am wiser. I have to revert this commit because it is indeed not completely researched and understood how all the games behave in situations with subareas without songlists.

Reply:
Using BREAKABLE_WEAPON was just a lazy fix, since perhaps another flag isn't needed. And to limit the change to bg1, due to the difference in iwd. #1854 has some good iwd2 RE linked. I suspect this actually used the same logic in all games.

Interestingly I don't see any master area checks in the code. The song header data also isn't inherited if empty:
https://github.com/alexbatalov/iwd2-re/blob/4fa36bcd6c33e0d35c9e273561aa683aece88490/src/CGameArea.cpp#L1884

So it might be that it just continued playing whatever was outside (all these -1 checks), however that can't explain why the bg1 area does have battle music.

Reply:
Thanks for the info and link. That's an interesting project.

Maybe the original Infinity Engine does different things for different games in this context. It could also be that the engine has the battle music hardcoded into the executable, i.e. in an area without songlist it just plays this one battle music if a battle starts.

It is possible that the fallback to the masterarea's songlist is actually not what's happening in the originals.

Hmm, in the end we don't really know. That's yet again a topic for wider and deeper research.

Reply:
strings doesn't suggest anything as bad as hardcoding battle music refs (which one would they pick anyway), while iwd2's version is straightforward https://github.com/alexbatalov/iwd2-re/blob/4fa36bcd6c33e0d35c9e273561aa683aece88490/src/CGameArea.cpp#L1333

Looking at why we added the game flag, the iwd problematic area from #1476 would pick up a song all the time. Even now that we conveniently store a close enough master area and could take it from there, that would still be bad, since that one does have a battlesong, but battles in the subarea don't have any. So we haven't learned anything new since then and the bg1 check should remain until further research shines more light on things.

Reply:
Ok, the commit is reverted. Comment and flag test are back in place.